scl/sumologic/sumologic.conf

```diff
@@ -0,0 +1,79 @@
+#############################################################################
+# Copyright (c) 2020 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# EXAMPLES:
+#
+# Just send all syslog to sumologic:
+#    log {
+#        source { system(); };
+#        destination { sumologic(token("USER-TOKEN-AS-PROVIDED-BY-sumologic")); };
+#    };
+#
+# With TLS encryption (make sure you trust the sumologic CA cert by putting it
+# to /etc/ssl, or create a separate CA directory):
+#
+#    log {
+#       destination {
+#                sumologic(token("USER-TOKEN-AS-PROVIDED-BY-sumologic") port(6514) transport(tls)
+#                       tls(peer-verify(required-trusted) ca-dir('/etc/ssl/certs'))
+#                );
+#       };
+#    };
+#
+#
+# Send JSON data:
+#    log {
+#        source { system(); };
+#        destination {
+#            sumologic(token("USER-TOKEN-AS-PROVIDED-BY-sumologic")
+#                   template("$(format-json --scope all-nv-pairs)"));
+#            );
+#        };
+#    };
+#
+#
+# Send already parsed apache logs to sumologic:
+#    log {
+#        source { file("/var/log/apache2/access.log" flags(no-parse)); };
+#        parser { apache-accesslog-parser(); };
+#        destination {
+#            sumologic(token("USER-TOKEN-AS-PROVIDED-BY-sumologic")
+#                   tag(apache)
+#                   template("$(format-json .apache.* timestamp=${ISODATE})"));
+#        };
+#    }
+#
+
+@requires json-plugin
+
+block destination sumologic(token(TOKEN)
+                         tag("tag")
+                         host('syslog.collection.ca.sumologic.com')
+                         port(6514)
+                         transport(tls)
+                         template("$MSG") ...) {
+  network("`host`" port(`port`) transport(`transport`)
+          template("<${PRI}>1 ${ISODATE} ${HOST} ${PROGRAM} ${PID} ${MSGID} [`token`@41123 tag=\"`tag`\"] `template`\n")
```

Comment:
Isn't `@41123` part of some specific token?
The [website](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/Cloud-Syslog-Source#send-data-to%C2%A0cloud-syslog-source-with-syslog-ng) says:
> Replace the sample token, E5kTyaEcth45/DU81M236oU4vM8j1ZaqTpWgjXB6lod7cFTeq09zzMn5ErmM0O/3@41123,
with your token.

Reply:
`The number 41123 in the token is the Sumo Private Enterprise Number (PEN). `

I think this key is fixed. So, maybe we can hard code it in the config.

Reply:
Hi @vatsal-encs ,
we shouldn't hard-code the PEN into the template as that is in the generated token as well.

This results into a duplicated PEn number, see my syslog-ng debug output:
`[2020-03-27T16:35:23.431747] Outgoing message; message='<13>1 2020-03-27T16:35:23+01:00 localhost da_logger 22918 - [token...zf/58Y@41123@41123 tag="tag"] bsd-source sent something\x0a'
`

Reply:
Hi @gaborznagy,

Okay, I will remove this hardcoded PEN.